lib/hsa/mcwamp_hsa.cpp

```diff
@@ -5846,24 +5868,11 @@ HSACopy::syncCopy() {
 // ----------------------------------------------------------------------
 
 extern "C" void *GetContextImpl() {
-  // If hc::printf is enabled, it must be initialized *after* the Kalmar::ctx is created.
-  // We only want to do this once, but the call to initPrintfBuffer will recurse back
-  // into this getter. We use TLS to make sure the same thread is the one recursing.
-  if (HCC_ENABLE_PRINTF) {
-    if (Kalmar::initPrintfBufferGuard) {
-      static std::once_flag flag;
-      std::call_once(flag, []() {
-        Kalmar::initPrintfBufferGuard = false;
-        Kalmar::ctx->initPrintfBuffer();
-        Kalmar::initPrintfBufferGuard = true;
-      });
-    }
-  }
-  return Kalmar::ctx;
+  return Kalmar::ctx();
 }
 
 extern "C" void ShutdownImpl() {
-  delete Kalmar::ctx;
+  delete Kalmar::__ctx;
```

Comment:
It seems that `Kalmar::__ctx` should be initialized to `nullptr` and then we should check whether it's still `nullptr` here before deleting it.